src/resolve.ts

```diff
@@ -11,7 +11,7 @@ import { fileURLToPath } from 'node:url';
  * @returns The resolved absolute path.
  */
 export function absolute(input: string, root?: string): string {
-	return isAbsolute(input) ? input : resolve(root || '.', input);
+	return isAbsolute(input) ? input : resolve(root ?? '.', input);
```

Comment:
Does expanding to that is slower or the engine is doing some work to make it faster than || that we don't know?
Because this is already a feature and its check is much more narrow than ||

Reply:
No. It's syntax only.

It's "narrow" is that it's doing more work to restrict/check types.

The syntax was added purely to prevent `0 | "" | false` accidental fall thru. It's literally `x != null ? ...`
??, ?= only hit the fallback on `null` and `undefined`

I'm writing `!= null` for brevity, but what it's actually expanding to is
```js
(x !== null && x !== undefined) ? x : "other"
```

which is worse ... or better.. depending on what you're valuing.

Reply:
If you are writing types like that you would expect ?? instead of ||
But generally i prefer narrowing the checks
But you can't be sure if engines dont opt this because it is a built in feature

Reply:
I assume the behavior you are mentioning is right if people are bundling to a JS version that this feature doesn't exist yet?